Dockerfile

```diff
@@ -1,25 +1,47 @@
-FROM registry.access.redhat.com/ubi8/python-39 as base
+FROM registry.access.redhat.com/ubi8/ubi-minimal AS build
 
 # Add application sources to a directory that the assemble script expects them
 # and set permissions so that the container runs without root access
 USER 0
 
-RUN dnf -y remove httpd-tools httpd-filesystem
+RUN microdnf install -y python39-devel postgresql-devel gcc && \
+    pip3 install virtualenv                                 && \
+    mkdir -p /opt/app-root                                  && \
+    chown 1001:0 /opt/app-root
 
-ADD app.py setup.py requirements.txt /tmp/src/
-ADD src /tmp/src/src
+USER 1001
+
+COPY --chown=1001:0 app.py setup.py requirements.txt /opt/app-root/
+COPY --chown=1001:0 src /opt/app-root/src
+
+WORKDIR /opt/app-root
+
+RUN virtualenv .                        && \
```

Comment:
We can do this as user 0 or as 1001, but as we're going to copy the files afterwards regarless, I think it's not needed to use the `chown` part of COPY. 

Also, we can first call `virtualenv` to create our directory, then use `WORKDIR` directive to set that virtual env dir. Then, copy files using relative routes, something like 

```
WORKDIR /opt/app-root/src/venv

COPY app.py setup.py requirements.txt .
COPY src ./src
```

Reply:
I don't want to tamper with the WORKDIR if possible, would be better to keep it static. But I'll adjust the paths relatively.